src/nodes/edge.h

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2025 Team Dissolve and contributors
+
+#pragma once
+
+#include "nodes/constants.h"
+#include "nodes/parameter.h"
+#include <string>
+
+// Forward Declarations
+class Graph;
+
+// Edge Definition
+struct EdgeDefinition
+{
+    std::string sourceNode, sourceOutput, targetNode, targetInput;
+    bool operator==(const EdgeDefinition &other) const = default;
+};
+
+// Edge
+class Edge
+{
+    private:
+    // Edge definition
+    EdgeDefinition definition_;
+
+    protected:
+    // The constructor is private because it can only be constructed by the factory method
+    Edge(Node *sourceNode, ParameterBase &sourceOutput, Node *targetNode, ParameterBase &targetInput);
+
+    private:
+    // Pointers to source and target nodes
+    Node *sourceNode_{nullptr};
+    Node *targetNode_{nullptr};
```

Comment:
```suggestion
    Node& sourceNode_;
    Node& targetNode_;
```

I rather liked having the nodes as references along with the parameters, for the same reasons given in the comment below.  While making this change will have some minor cascading effects with replacing `->` with `.`, I don't think it should break anything major.

Reply:
I moved to pointers because (I think) while I was starting to combine the `Edge` and `ParameterLink` classes I thought I needed to, but you are absolutely right that use of references still stands as the best option (and there's no technical reason why this won't work) so I will revert this.